### ISSUE TITLE
Wrong sort-comp parsing.

### ISSUE THREAD

**Issue by kasperekt** [yannickcr/eslint-plugin-react#149]
While having this code:

```js
const Input = React.createClass({
  componentDidMount() {
    console.log('mounted');
  },

  getInitialState() {
    return {
      a: 1,
    };
  },
});
```

I have this ruleset:

```json
"react/sort-comp": [2, {
      "order": [
        "displayName",
        "propTypes",
        "contextTypes",
        "childContextTypes",
        "mixins",
        "statics",
        "defaultProps",
        "getDefaultProps",
        "getInitialState",
        "getChildContext",
        "componentWillMount",
        "componentDidMount",
        "componentWillReceiveProps",
        "shouldComponentUpdate",
        "componentWillUpdate",
        "componentDidUpdate",
        "componentWillUnmount",
        "/^on.+$/",
        "/^get.+$/",
        "/^render.+$/",
        "render"
      ]
    }]
```

It shows no errors, where it should tell me to move `componentDidMount` below `getInitialState`

**Comment by yannickcr** [comment]
In your configuration the `/^get.+$/`RegEx allows the `getInitialState`function to be placed bellow `componentDidMount`.

I'm not a RegEx expert, but I think the following RegEx should fix the problem:

`/^get(?!(InitialState$|DefaultProps$|ChildContext$)).+$/`

It should only allow methods names starting with get and __not__ followed by `InitialState`, `DefaultProps` or `ChildContext`.

**Comment by kasperekt** [comment]
Yep, that was the case. Thank you @yannickcr
